sway-core/src/language/ty/expression/expression_variant.rs

```diff
@@ -51,6 +51,7 @@ pub enum TyExpressionVariant {
         struct_name: Ident,
         fields: Vec<TyStructExpressionField>,
         span: Span,
+        type_binding: TypeBinding<()>,
```

Comment:
Would it makes sense to combine the `struct_name` and `type_binding` fields?

Reply:
I've pondered this but I think it makes more sense to keep them separate, `TypeBinding<()>` is pretty clearly meant to be consumed by the LSP as used so far and making generic over random other related stuff seems like a bad idea.

In fact, I've been thinking about doing away with the generic argument for it altogether and splitting  `TypeBinding<()>`, `TypeBinding<CallPath<(TypeInfo, Span)>>` and `TypeBinding<CallPath>` into properly separate types or at least separate struct fields. The composition doesn't seem very ergonomic to me as it is.
What do you think?

Reply:
> In fact, I've been thinking about doing away with the generic argument for it altogether and splitting TypeBinding<()>, TypeBinding<CallPath<(TypeInfo, Span)>> and TypeBinding<CallPath> into properly separate types or at least separate struct fields. The composition doesn't seem very ergonomic to me as it is.
What do you think?

The original design of `TypeBinding` was intended to be modular and """left-associated""". So instead of both the `StructExpression` variant, the `FunctionApplication` variant, etc, keeping track of the fact that their calling conventions could take a turbofish, this would be """left-associated""" with the """name""" field. For this reason, I prefer `TypeBinding<CallPath>` over a specific `type_binding` field on `StructExpression`.

But you're right, `TypeBinding` is not very ergonomic. I'm wondering if there is a better compromise squashing `TypeBinding` into `CallPath`? I'm curious to look at other designs---take a crack at it if you're interested!

If you do decide to rework `TypeBinding`, I recommend that we merge this PR and that you open a new issue and a new PR for this specifically.

Reply:
Agreed, I think this is out of scope here but I'll definitely look into refactoring it in a later PR.